File: food-order-service/src/models/Restaurant.ts
import { makeObservable, observable, computed, action, runInAction } from 'mobx';
import { MenuStore } from '../stores/MenuStore';

export interface IRestaurantItem {
  id: string;
  name: string;
  description: string;
  workingHours: string;
  address: string;
  telephoneNumber: string;
}

export class Restaurant implements IRestaurantItem {
  @observable public data: IRestaurantItem;
  @observable public id: string;
  @observable public name: string;
  @observable public description: string;
  @observable public workingHours: string;
  @observable public address: string;
  @observable public telephoneNumber: string;
  @observable public MenuStore: MenuStore = new MenuStore();

  public constructor(data: IRestaurantItem) {
    makeObservable(this);
    this.initFromData(data);
  }

  @action
  public initFromData(data: IRestaurantItem) {
    this.data = data;
    this.id = data.id;
    this.name = data.name;
    this.description = data.description;
    this.workingHours = data.workingHours;
    this.address = data.address;
    this.telephoneNumber = data.telephoneNumber;
  }
}

File: foodfast-project/API/API.Data/Models/Restaurant.cs
using System;
using System.Collections.Generic;

namespace API.Data.Models;

public partial class Restaurant
{
    public long Id { get; set; }

    public string Name { get; set; } = null!;

    public string Description { get; set; } = null!;

    public string WorkingHours { get; set; } = null!;

    public string Address { get; set; } = null!;

    public string PhoneNumber { get; set; } = null!;

    public int Available { get; set; }
}
